### PR TITLE
openstack Nova region

### LINKED ISSUE #3336: Add a `region` arg to `OpenStackLatentWorker`
Would be nice to allow passing a region_name to novaclient in `OpenStackLatentWorker`.

### PR DESCRIPTION
**Pull request by cmouse**
## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

closes #3336

### CODE REVIEW

File: master/buildbot/worker/openstack.py [hunk 1]
```diff
@@ -88,6 +89,8 @@ def __init__(self, name, password,
             self.novaclient = self._constructClient(
                 client_version, os_username, os_password, os_tenant_name,
                 os_auth_url)
+            if region is not None:
+                self.novaclient.client.region = region
```
Comment (seankelly) [inline master/buildbot/worker/openstack.py]: The [docs](https://docs.openstack.org/python-novaclient/latest/reference/api/novaclient.v2.client.html) show the Client supporting a `region_name` parameter.

Comment (cmouse) [inline master/buildbot/worker/openstack.py]: thanks.

File: master/buildbot/test/unit/test_worker_openstack.py [hunk 2]
```diff
@@ -71,6 +71,11 @@ def test_constructor_minimal(self):
         self.assertEqual(bs.block_devices, None)
         self.assertIsInstance(bs.novaclient, novaclient.Client)
 
+    def test_constructor_region(self):
+        bs = openstack.OpenStackLatentWorker(
+            'bot', 'pass', region="test-region", **self.bs_image_args)
+        self.assertEqual(bs.novaclient.client.region, "test-region")
```
Comment (tardyp) [inline master/buildbot/test/unit/test_worker_openstack.py]: region_name

**Comment by cmouse** [comment]
rebased over master to trigger build

**Pull request merged**